test/options_description_test.cpp

```diff
@@ -315,6 +316,20 @@ void test_value_name()
    );
 }
 
+void test_backwards_compatibility_with_raw_pointer()
```

Comment:
I thought it was a good idea to add a test case to test the backwards compatibility.